pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -42,21 +42,21 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 
 	// First try to calculate changes will update references on all resources
 	// (which includes versioned and non-versioned resources)
-	_, _, err := d.addChanges(newRs, existingRsGrouped)
+	_, _, _, err := d.addChanges(newRs, existingRsGrouped)
 	if err != nil {
 		return nil, err
 	}
 
 	// Since there might have been circular dependencies;
 	// second try catches ones that werent changed during first run
-	addChanges, alreadyAdded, err := d.addChanges(newRs, existingRsGrouped)
+	addChanges, alreadyAdded, alreadyAddedKeepOp, err := d.addChanges(newRs, existingRsGrouped)
```

Comment:
Maybe use the same map (`alreadyAdded`) with `Change` as a value instead of `Resource`?